Feature/endpoint crear turnos

jeremanuel opened the pull request:
## Descripción

Se hizo todo el camino de datos para la creación de turnos. (presentation -> domain -> infrastructure).

## Capturas de pantalla (si es aplicable)

![image](https://github.com/jeremanuel/turni/assets/60302223/42fe308c-1d48-4581-89c6-4a91b5d4b35f)

Por ahora agregue un botón simplemente ahí abajo.

## Comprobación

Antes de enviar este PR, por favor, asegúrate de que has realizado las siguientes comprobaciones:

- [x] Me he asignado como responsable del PR.
- [x] He añadido etiquetas si es necesario.
- [x] He probado este código en mi entorno local.
- [x] He agregado documentación si es necesario.
- [ ] He agregado test unitarios.

## Notas adicionales

Hay que debatir @JoaquinMastropierro @AgustinIMassa qué validaciones vamos a hacer al guardar. 

Por ejemplo, 

comprobar que los turnos que vamos a agregar, no se pisen con turnos existentes?.

Y tambien validaciones locales, más simples, osea, no permitir que se pisen dentro de los que vamos a guardar actualmente.
 
También hay que ver que hacer con el tipo de dato "duration" ya que no estamos pudiendo guardar ni fetchear a través de prisma. 

Tal vez convenga simplemente guardar un string que represente la duración? EJ: "01:30"

## Merge Strategy

| From | To | Strategy |
|--------|--------|--------|
| Feature | Develop | Squash&Merge |
| Release | Master | Merge Commit |
| Hotfix | Master | Squash&Merge |

jereTASR committed:
Agregue un metodo en el tipo de dato TimeInterval, para generar un arreglo con las fechas que pertenecen al intervalo

jereTASR committed:
Cree el camino, desde presentation hasta infrastructure para la creacion de turnos

jereTASR committed:
elimine clase vieja

jereTASR committed:
Cree un nuevo evento, para reiniciar la visualizacion actual, util para cuando navego desde la creacion de turnos hacia atras, para que se vea actualizado el panel

jereTASR committed:
Puse la APP en modo oscuro

jereTASR committed:
Cambie el color de el divider que representa el dia actual, para que se visualize bien en modo oscuro

jereTASR committed:
Agregue bloc listener, para que cuando guardo turnos, lo represento en el state, y al escuchar ese cambio en la vista, navego hacia el session_manager

AgustinIMassa commented:
Si tema validaciones en local a priori no nos afecta por ahora, despues lo mas probable es que si con mas admins para una misma partition

El duration hay que castearlo a text(string) en los rawquerys para la obtencion, no se como se hara en el guardado, pero para obtenerlos probe y funciona. Hay un ejemplito ::text

jereTASR committed:
Agregue FlutterLocalization, para indicar el idioma de la app, y que el calendario se visualize en español

jereTASR committed:
Cree un shellroute en el session manager, para tener rutas que solo modifican una parte del sessionManager (la ultima columna)

jereTASR committed:
inicializacion de flutterlocalizations

jereTASR committed:
Mejoras visuales en sidebar. elimine archivo viejo

jereTASR committed:
Mejoras visuales en sidebar

jereTASR committed:
Mejora en carrousel para el scroll, mejora en Agenda

jereTASR committed:
elimine key

jereTASR committed:
Hice que el sessionManager reciba el sideChild por parametro, para utilizarlo como shellroute

jereTASR committed:
correcciones simples

jereTASR committed:
Abstraccion de multiples componentes en widgets independientes.

Implementacion de ShellRouter en SessionManager, para cambiar la ultima columna en base a la ruta
